src/request.c

```diff
@@ -156,27 +156,113 @@ request_authorize_callback (GDBusInterfaceSkeleton *interface,
   return TRUE;
 }
 
+/* This is a bit ugly - we need to know where the options vardict is
+ * in the parameters for each request. Instead of inventing some
+ * complicated mechanism for each implementation to provide that
+ * information, just hardcode it here for now.
+ */
+static const char *
+get_token (GDBusMethodInvocation *invocation)
+{
+  const char *interface;
+  const char *method;
+  GVariant *parameters;
+  g_autoptr(GVariant) options = NULL;
+  const char *token = "t";
```

Comment:
No, I actually mean "handles". That is what we call the out argument everywhere: @handle.